File: app/src/main/java/io/mgba/UI/Activities/Interfaces/IMainView.java
package io.mgba.UI.Activities.Interfaces;

import android.support.v7.app.AppCompatActivity;
import com.arlib.floatingsearchview.suggestions.model.SearchSuggestion;
import com.mikepenz.aboutlibraries.LibsBuilder;
import java.util.List;
import io.mgba.Data.Database.Game;

public interface IMainView {
    void clearSuggestions();
    void showSuggestions(List<? extends SearchSuggestion> list);
    void startAboutPanel(LibsBuilder aboutPanel);
    void startActivityForResult(Class<? extends AppCompatActivity> activity, int code);
    void showProgress();
}

File: app/src/main/java/io/mgba/ui/activities/interfaces/IMainView.kt
package io.mgba.ui.activities.interfaces


import com.arlib.floatingsearchview.suggestions.model.SearchSuggestion
import com.mikepenz.aboutlibraries.LibsBuilder
import androidx.appcompat.app.AppCompatActivity

interface IMainView {
    fun clearSuggestions()
    fun showSuggestions(list: List<SearchSuggestion>)
    fun startAboutPanel(aboutPanel: LibsBuilder)
    fun startActivityForResult(activity: Class<out AppCompatActivity>, code: Int)
    fun showProgress()
}
